### DISCUSSION TITLE
E2E/integration test needed!

### DISCUSSION THREAD

**Discussion by dadiorchen** [Greenstand/treetracker-web-map-client#245] [General]
Our new version of web map caused the crash of timeline: https://github.com/Greenstand/treetracker-web-map-client/issues/244

How could we avoid this?

Is this indicating that the necessary of end to end test? or integration test? If we adapt e2e test, how could we deal with the data? How do we maintain the env/database needed by the e2e tests? If we use integration test to cover features like: timeline filter, how do we mock the data? It's a bit challenging testing map applications.

**Comment by dadiorchen** [comment]
@michaltrunek what's your opnion?

**Reply by michaltrunek** [reply]
Hi Dadi, if we are retrieving data from API for the timeline I would use that and check we are getting some data. I can see that the timeline is still not working but there can be also e2e test for this. 

We can do similar test case for example:
1. Open https://map.treetracker.org/?timeline=27-10-2018_03-11-2021 - where the last part of the URL is the timeline filter
2. Check that there are some trees/badges with numbers in the UI

Let me know what you think ;)

**Reply by dadiorchen** [reply]
By doing what you said, it's impossible to find this bug, let me explain to you the case:
Previously, we use the timeline slider on the client-side to select the time range, then request the server API for data. the correct request URL should be like: http://xxxxx/?timeline=2018-01-01_2018-01-31
Then our developer made some modification to change the date format on this client-side (from 2018-01-01 to 01-01-2018 to follow a better convention), without realizing that it will change to request to the API, so the server-side don't know the change, so the API request failed, actually, this code change just care about the UI appearance (the data format) for end-user, shouldn't change the request URL to the server.
So if you directly open the URL with timeline parameters, it's possible that we will get the correct result, but when the user operate the timeline filter on the browser, they might get wrong result.

**Reply by michaltrunek** [reply]
I see, then there you can change navigation to the specific timeline for drag and drop - interaction with the real timeline. Drag and drop action should do the magic.
